tests/qos/test_pfc_counters.py

```diff
@@ -182,40 +182,40 @@ def run_test(fanouthosts, duthost, conn_graph_facts, fanout_graph_facts, leaf_fa
                     assert pfc_rx[intf]['Rx'][i] == '0'
 
 
-def test_pfc_pause(fanouthosts, duthosts, rand_one_dut_hostname,
```

Comment:
is the failure observed when supervisor was choosen?

Reply:
No, you can see the details on this ticket here: https://msazure.visualstudio.com/One/_sprints/taskboard/SONiC%20High%20Gear/One/Dilithium/CY24Q3/2Wk/2Wk12%20(Sep%2008%20-%20Sep%2021)?System.AssignedTo=austinpham%40microsoft.com&workitem=29400374

The fanout_graph_facts selected a different line card comparing to the linecard is being tests. Therefore it does not have the correct devices while during the test simply because fanout_graph_facts is reading something else.

In the failure case for example fanout is reading lc2-1 while we're doing the test for lc4-1 whereas the random dut should be consistent across graph_facts and the test

Reply:
As discussed.  The reason for this issue is: in test case itself, it use `rand_one_dut_hostname` and fanout_graph_facts() uses `rand_one_tgen_dut_hostname`.  These two fixtures may return different dut when multiple devices are there, hence causing failure.

Take this as an example: 
```
def test_pfc_pause(fanouthosts, duthosts, rand_one_dut_hostname,
                   conn_graph_facts, fanout_graph_facts, leaf_fanouts):          # noqa F811
    """ @Summary: Run PFC pause frame (pause time quanta > 0) tests """
    duthost = duthosts[rand_one_dut_hostname]
```
The `fanout_graph_facts` may use lc4, but `rand_one_dut_hostname` may use lc2. Hence the dut doesn't match.

This change looks good to me. 
We will need the same change for `qos/test_pfc_pause.py`  too. Seems these are the only two places which used both `fanout_graph_facts` and `rand_one_dut_hostname`.